core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -562,7 +577,7 @@ protected QB copyQuery(QB query) throws IOException {
      * @return a new {@link QueryShardContext} based on the base test index and queryParserService
      */
     protected static QueryShardContext createShardContext() {
-        QueryShardContext queryCreationContext = new QueryShardContext(queryParserService);
+        QueryShardContext queryCreationContext = queryShardContext.clone();
```

Comment:
talking about not using clone here, but simply doing `new QueryShardContext(...all the needed deps)`